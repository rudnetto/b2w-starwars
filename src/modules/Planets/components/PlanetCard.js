// @flow
import React from 'react';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import Components from './Styled';
import { Planet } from '../Models';

type Props = {
  planet?: Planet;
}

const PlanetCard = (props: Props) => {
  return (
    <Components.Card raised>
      { props.planet && 
        <React.Fragment>
          <CardHeader title={props.planet.name} />
          <CardContent>
            <Typography color="inherit" variant="subheading">
              <b>Population:</b> {props.planet.population}
            </Typography>
            <Typography color="inherit" variant="subheading">
              <b>Climate:</b> {props.planet.climate}
            </Typography>
            <Typography color="inherit" variant="subheading">
            <b>Terrain:</b> {props.planet.terrain}
            </Typography>
            <Typography color="inherit" variant="subheading">
            This planet was featured in <b>{props.planet.filmsFeatured}</b> film(s).
            </Typography>
          </CardContent>
        </React.Fragment>
      }
      {
        !props.planet &&
        <React.Fragment>
          <CardHeader title="Error" />
          <CardContent>
            <Typography color="inherit" variant="subheading">
              Something happened when reaching the server. Please try again.
            </Typography>
          </CardContent>
        </React.Fragment>
      }
    </Components.Card>
  );
}

export default PlanetCard;
