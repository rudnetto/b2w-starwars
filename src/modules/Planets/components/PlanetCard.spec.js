import React from 'react';
import { shallow } from 'enzyme';
import PlanetCard from './PlanetCard';

const planetMock = {
  name: 'Tatooine',
  population: '120000',
  terrain: 'Dessert',
  climate: 'Arid',
  filmsFeatured: 5,
};

describe('PlanetCard test suite', () => {
  it('should render', () => {
    const element = shallow(<PlanetCard planet={planetMock} />);
    expect(element).toMatchSnapshot();
  });

  it('should render error', () => {
    const element = shallow(<PlanetCard planet={null} />);
    expect(element).toMatchSnapshot();
  });
});
