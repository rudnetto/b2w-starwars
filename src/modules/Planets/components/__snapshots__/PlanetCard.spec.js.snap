// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetCard test suite should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard
    planet={
        Object {
            "climate": "Arid",
            "filmsFeatured": 5,
            "name": "Tatooine",
            "population": "120000",
            "terrain": "Dessert",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown>
          <WithStyles(CardHeader)
                    title="Tatooine"
          />
          <WithStyles(CardContent)>
                    <WithStyles(Typography)
                              color="inherit"
                              variant="subheading"
                    >
                              <b>
                                        Population:
                              </b>
                               120000
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              color="inherit"
                              variant="subheading"
                    >
                              <b>
                                        Climate:
                              </b>
                               Arid
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              color="inherit"
                              variant="subheading"
                    >
                              <b>
                                        Terrain:
                              </b>
                               Dessert
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              color="inherit"
                              variant="subheading"
                    >
                              This planet was featured in
                              <b>
                                         5 
                              </b>
                              film(s).
                    </WithStyles(Typography)>
          </WithStyles(CardContent)>
</Unknown>,
        false,
      ],
      "raised": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WithStyles(CardHeader)
              title="Tatooine"
/>,
            <WithStyles(CardContent)>
              <WithStyles(Typography)
                            color="inherit"
                            variant="subheading"
              >
                            <b>
                                          Population:
                            </b>
                             120000
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            color="inherit"
                            variant="subheading"
              >
                            <b>
                                          Climate:
                            </b>
                             Arid
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            color="inherit"
                            variant="subheading"
              >
                            <b>
                                          Terrain:
                            </b>
                             Dessert
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            color="inherit"
                            variant="subheading"
              >
                            This planet was featured in
                            <b>
                                           5 
                            </b>
                            film(s).
              </WithStyles(Typography)>
</WithStyles(CardContent)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "title": "Tatooine",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  color="inherit"
                  variant="subheading"
>
                  <b>
                                    Population:
                  </b>
                   120000
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  color="inherit"
                  variant="subheading"
>
                  <b>
                                    Climate:
                  </b>
                   Arid
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  color="inherit"
                  variant="subheading"
>
                  <b>
                                    Terrain:
                  </b>
                   Dessert
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  color="inherit"
                  variant="subheading"
>
                  This planet was featured in
                  <b>
                                     5 
                  </b>
                  film(s).
</WithStyles(Typography)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <b>
                      Population:
</b>,
                    " 120000",
                  ],
                  "color": "inherit",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Population:",
                    },
                    "ref": null,
                    "rendered": "Population:",
                    "type": "b",
                  },
                  " 120000",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <b>
                      Climate:
</b>,
                    " Arid",
                  ],
                  "color": "inherit",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Climate:",
                    },
                    "ref": null,
                    "rendered": "Climate:",
                    "type": "b",
                  },
                  " Arid",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <b>
                      Terrain:
</b>,
                    " Dessert",
                  ],
                  "color": "inherit",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Terrain:",
                    },
                    "ref": null,
                    "rendered": "Terrain:",
                    "type": "b",
                  },
                  " Dessert",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "This planet was featured in",
                    <b>
                       5 
</b>,
                    "film(s).",
                  ],
                  "color": "inherit",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": Array [
                  "This planet was featured in",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " 5 ",
                    },
                    "ref": null,
                    "rendered": " 5 ",
                    "type": "b",
                  },
                  "film(s).",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown>
            <WithStyles(CardHeader)
                        title="Tatooine"
            />
            <WithStyles(CardContent)>
                        <WithStyles(Typography)
                                    color="inherit"
                                    variant="subheading"
                        >
                                    <b>
                                                Population:
                                    </b>
                                     120000
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                    color="inherit"
                                    variant="subheading"
                        >
                                    <b>
                                                Climate:
                                    </b>
                                     Arid
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                    color="inherit"
                                    variant="subheading"
                        >
                                    <b>
                                                Terrain:
                                    </b>
                                     Dessert
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                    color="inherit"
                                    variant="subheading"
                        >
                                    This planet was featured in
                                    <b>
                                                 5 
                                    </b>
                                    film(s).
                        </WithStyles(Typography)>
            </WithStyles(CardContent)>
</Unknown>,
          false,
        ],
        "raised": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WithStyles(CardHeader)
                title="Tatooine"
/>,
              <WithStyles(CardContent)>
                <WithStyles(Typography)
                                color="inherit"
                                variant="subheading"
                >
                                <b>
                                                Population:
                                </b>
                                 120000
                </WithStyles(Typography)>
                <WithStyles(Typography)
                                color="inherit"
                                variant="subheading"
                >
                                <b>
                                                Climate:
                                </b>
                                 Arid
                </WithStyles(Typography)>
                <WithStyles(Typography)
                                color="inherit"
                                variant="subheading"
                >
                                <b>
                                                Terrain:
                                </b>
                                 Dessert
                </WithStyles(Typography)>
                <WithStyles(Typography)
                                color="inherit"
                                variant="subheading"
                >
                                This planet was featured in
                                <b>
                                                 5 
                                </b>
                                film(s).
                </WithStyles(Typography)>
</WithStyles(CardContent)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "title": "Tatooine",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Typography)
                    color="inherit"
                    variant="subheading"
>
                    <b>
                                        Population:
                    </b>
                     120000
</WithStyles(Typography)>,
                  <WithStyles(Typography)
                    color="inherit"
                    variant="subheading"
>
                    <b>
                                        Climate:
                    </b>
                     Arid
</WithStyles(Typography)>,
                  <WithStyles(Typography)
                    color="inherit"
                    variant="subheading"
>
                    <b>
                                        Terrain:
                    </b>
                     Dessert
</WithStyles(Typography)>,
                  <WithStyles(Typography)
                    color="inherit"
                    variant="subheading"
>
                    This planet was featured in
                    <b>
                                         5 
                    </b>
                    film(s).
</WithStyles(Typography)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <b>
                        Population:
</b>,
                      " 120000",
                    ],
                    "color": "inherit",
                    "variant": "subheading",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Population:",
                      },
                      "ref": null,
                      "rendered": "Population:",
                      "type": "b",
                    },
                    " 120000",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <b>
                        Climate:
</b>,
                      " Arid",
                    ],
                    "color": "inherit",
                    "variant": "subheading",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Climate:",
                      },
                      "ref": null,
                      "rendered": "Climate:",
                      "type": "b",
                    },
                    " Arid",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <b>
                        Terrain:
</b>,
                      " Dessert",
                    ],
                    "color": "inherit",
                    "variant": "subheading",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Terrain:",
                      },
                      "ref": null,
                      "rendered": "Terrain:",
                      "type": "b",
                    },
                    " Dessert",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "This planet was featured in",
                      <b>
                         5 
</b>,
                      "film(s).",
                    ],
                    "color": "inherit",
                    "variant": "subheading",
                  },
                  "ref": null,
                  "rendered": Array [
                    "This planet was featured in",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": " 5 ",
                      },
                      "ref": null,
                      "rendered": " 5 ",
                      "type": "b",
                    },
                    "film(s).",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PlanetCard test suite should render error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard
    planet={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        null,
        <Unknown>
          <WithStyles(CardHeader)
                    title="Error"
          />
          <WithStyles(CardContent)>
                    <WithStyles(Typography)
                              color="inherit"
                              variant="subheading"
                    >
                              Something happened when reaching the server. Please try again.
                    </WithStyles(Typography)>
          </WithStyles(CardContent)>
</Unknown>,
      ],
      "raised": true,
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WithStyles(CardHeader)
              title="Error"
/>,
            <WithStyles(CardContent)>
              <WithStyles(Typography)
                            color="inherit"
                            variant="subheading"
              >
                            Something happened when reaching the server. Please try again.
              </WithStyles(Typography)>
</WithStyles(CardContent)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "title": "Error",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Typography)
                color="inherit"
                variant="subheading"
>
                Something happened when reaching the server. Please try again.
</WithStyles(Typography)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Something happened when reaching the server. Please try again.",
                "color": "inherit",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "Something happened when reaching the server. Please try again.",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          null,
          <Unknown>
            <WithStyles(CardHeader)
                        title="Error"
            />
            <WithStyles(CardContent)>
                        <WithStyles(Typography)
                                    color="inherit"
                                    variant="subheading"
                        >
                                    Something happened when reaching the server. Please try again.
                        </WithStyles(Typography)>
            </WithStyles(CardContent)>
</Unknown>,
        ],
        "raised": true,
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WithStyles(CardHeader)
                title="Error"
/>,
              <WithStyles(CardContent)>
                <WithStyles(Typography)
                                color="inherit"
                                variant="subheading"
                >
                                Something happened when reaching the server. Please try again.
                </WithStyles(Typography)>
</WithStyles(CardContent)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "title": "Error",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)
                  color="inherit"
                  variant="subheading"
>
                  Something happened when reaching the server. Please try again.
</WithStyles(Typography)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Something happened when reaching the server. Please try again.",
                  "color": "inherit",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": "Something happened when reaching the server. Please try again.",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
